plugin.video.slingtv/resources/lib/slingtv.py

```diff
@@ -0,0 +1,789 @@
+                        'locale': disney_info.get('disney_locale', ''),
+                        'token': '%s_%s_%s' % (sha1_user_id, utc_datetime, signature),
+                        'token_type': 'offsite_dish_ott',
+                        'user_id': sha1_user_id,
+                        'video_type': 'live',
+                        'zip_code': USER_ZIP
+                    }
+                    service_url = disney_info['disney_stream_service_url']
+                    log('service url %s' % service_url)
+                    payload = ''
+                    for key in params.keys():
+                        payload += '%s=%s&' % (key, params[key])
+                    payload = payload[:-1]
+                    response = requests.post(service_url, headers=stream_headers, data=payload, verify=VERIFY)
+                    if response.status_code == 200:
+                        log(str(response.text))
+                        session_xml = xmltodict.parse(response.text)
+                        service_stream = session_xml['playmanifest']['channel']['assets']['asset']['#text']
+                        log('resolverURL, XML Stream: ' + str(service_stream))
+                        mpd_url = service_stream
+                        license_key = ''
+
+        asset_id = ''
+        if 'entitlement' in video and 'asset_id' in video['entitlement']:
+            asset_id = video['entitlement']['asset_id']
+
+        return mpd_url, license_key, asset_id
+
+        
+    def playVideo(self, name, url, liz=None):
+        log('playVideo, url = ' + url)
+        try:
+            url, license_key, external_id = self.resolverURL(url)
+        except:
+            license_key = ''
+            external_id = ''
+        if 'mpd' in url:
+            is_helper = inputstreamhelper.Helper('mpd', drm='widevine')
+            if not is_helper.check_inputstream():
+                sys.exit()
+            liz = xbmcgui.ListItem(name, path=url)
+            liz.setProperty('inputstreamaddon', 'inputstream.adaptive')
+            liz.setProperty('inputstream.adaptive.manifest_type', 'mpd')
+            liz.setProperty('inputstream.adaptive.stream_headers', 'User-Agent='+USER_AGENT)
+            if license_key != '':
+                liz.setProperty('inputstream.adaptive.license_type', 'com.widevine.alpha')
+                liz.setProperty('inputstream.adaptive.license_key', license_key)
+            liz.setMimeType('application/dash+xml')
+            liz.setContentLookup(False)
+            xbmcplugin.setResolvedUrl(int(self.sysARG[1]), True, liz)
+        else:
+            liz = xbmcgui.ListItem(name, path=url)
+            xbmcplugin.setResolvedUrl(int(self.sysARG[1]), True, liz)
+
+        # Hack to fix 6 channel audio causing buffering issues
+        while not xbmc.Player().isPlayingVideo():
+            xbmc.Monitor().waitForAbort(0.25)
+
+        if xbmc.Player().isPlayingVideo() and len(xbmc.Player().getAvailableAudioStreams()) > 1:
```

Comment:
If this is a bug in Kodi shouldn't it be reported instead of just working around it?